bokehjs/src/coffee/common/embed.coffee

```diff
@@ -8,6 +8,18 @@ HasProperties = require "./has_properties"
 {pull_session} = require "./client"
 {Promise} = require "es6-promise"
 
+_handle_notebook_comms = (msg) ->
+  logger.debug("handling notebook comms")
+  doc = _.values(@)[0] # should only be one
```

Comment:
Adding comms creation to the middle of `_standalone_docs_json_and_render_items` seems a bit strange but I don't see anywhere else to do it. Also, `notebook_div` currently only always sends one doc, it does:
```
item = render_items[0]

div = _div_for_render_item(item)
```